[config] Cleanup and SafetyRules prep

davidiw opened the pull request:
This PR will serve to major purposes:
- Eliminating cruft due to genesis being the source of truth (recent work) and cruft in general
- Moving appropriate data into a place that makes sense from a SafetyRules perspective

The specifics:
- Cleaning up dynamic-config-builder junk
- Moving consensus_keypair from consensus config to test config
- Eliminating consensus_peers from consensus_config
- Eliminating the need for network_peers for validator networks
- Generating a ValidatorSet during validator generation rather than network / consensus peer configs
- Deleting consensus peer config

Testing relied primarily on integration tests and leveraging the local dynamic run of docker images.

bors-libra commented:
:umbrella: The latest upstream changes (presumably 0f97648bd9df4fa8e5da2eadc34d164e75727529) made this pull request unmergeable. Please resolve the merge conflicts.

dmitri-perelman commented:
@bors-libra delegate+

bors-libra commented:
:v: @davidiw can now approve this pull request

davidiw commented:
@dmitri-perelman, thanks for the review!

Updated to include testing method. I fixed our local running of docker images.
The smoke tests are actually really good at testing these type of things as well.
I did try to use the cluster testing but hit some issues with the environment setup.

davidiw commented:
@bors-libra r=dmitri-perelman

bors-libra commented:
:pushpin: Commit 0dd9098 has been approved by `dmitri-perelman`

<!-- @bors-libra r=dmitri-perelman 0dd9098570afa44ad4cbbf7cb28403b17e5ee774 -->

bors-libra commented:
:hourglass: Testing commit 0dd9098570afa44ad4cbbf7cb28403b17e5ee774 with merge a7ef849dff190466ef10518157ded26a31d278ab...

bors-libra commented:
:sunny: Test successful - [checks-circle_commit_workflow](https://circleci.com/workflow-run/46e27823-c92c-4011-bbf5-57becc5ba467?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link)
Approved by: dmitri-perelman
Pushing a7ef849dff190466ef10518157ded26a31d278ab to master...

aching commented:
Btw, looks like a nice cleanup, thanks!

davidiw commented on config/src/config/consensus_config.rs:
```diff
@@ -1,69 +1,40 @@
+        self.safety_rules.set_data_dir(data_dir);
+    }
+}
+
```
That's true, I think eventually I'm just going to move safety rules to being top level.
